src/jit/compiler.h

```diff
@@ -1487,8 +1471,8 @@ struct fgArgTabEntry
     bool _isSplit : 1; // True when this argument is split between the registers and OutArg area
 #endif                 // FEATURE_ARG_SPLIT
 #ifdef FEATURE_HFA
-    bool _isHfaArg : 1;    // True when the argument is an HFA type.
-    bool _isDoubleHfa : 1; // True when the argument is an HFA, with an element type of DOUBLE.
+    bool          _isHfaArg : 1;   // True when the argument is an HFA type.
+    unsigned char _hfaLogSize : 2; // The base 2 log of the HFA element size in 4-byte units (e.g. 0 for TYP_FLOAT)
```

Comment:
Isn't there other terminology that could be used here? "size" most of the type refers to byte size. In call morphing code "size" also refers to the number of slots. And now here it refers to the number of 4-byte units.

Reply:
How about `_hfaLogWidth`? Or "_hfaLogRelSize" for relative size?

Reply:
Hmm, width is also used in some cases to denote byte size (some block ops code does that). But at least it's not used within `LclVarDsc` (which has size and exactSize) so it's a better alternative.

Though I'm not sure why this is needed to begin with. Is it just to "compress" the information down to 2 bits in the bitfield?

Reply:
Yes, it's just to keep the information small as opposed to making it a `var_types` field, or creating a new enum.

Reply:
Hmm, then leave it as it is. I have a feeling that this HFA info will end up in the `ClassLayout` thing I have in #21705 sooner or later.

Reply:
In looking at my response, I'm thinking a new enum would be better - it could be called something like `HfaElemKind` and have values `HFA_ELEM_FLOAT`, `HFA_ELEM_DOUBLE` and `HFA_ELEM_VECTOR128`. And then we could rely on the C++ compiler to generate good code rather than trying to pre-optimize by using shifts. (I try to make it a principle to rely on the code generator and not pre-optimize, but I don't always manage to do so!)

Reply:
Right, an enum should be fine as long as it has underlying type `unsigned char` to match the other bitfield members. Otherwise AFAIR VC++ does not pack data member of different types into the same byte of the bitfield.